### DISCUSSION TITLE
Better folder preview

### DISCUSSION THREAD

**Discussion by vladstudio** [mikekazakov/nimble-commander#252] [Ideas]
Right now, when I hit "Show preview" on a folder, it looks like this:

<img width="1136" alt="image" src="https://github.com/mikekazakov/nimble-commander/assets/914320/0e7961f4-9f03-4fd6-8746-22bf56401732">

Not very helpful! Do you think it could include:
- calculated size (as requested [here](https://github.com/mikekazakov/nimble-commander/issues/114) )
- maybe even folder listing, so that I could browse folder on one panel, and quickly view their contents on the other panel, similar to this:

https://github.com/mikekazakov/nimble-commander/assets/914320/e3996701-100f-4e56-83f7-bd9c7e659ce3

**Comment by lcerman** [comment]
Hello, I could even imagine to display more information than just the calculated size, e.g., something like this:
![image](https://github.com/mikekazakov/nimble-commander/assets/20700132/330d80ca-2a3e-4fb4-8d7c-53e4e9bb945b)

(this screenshot comes from Altap Salamander - Calculate Occupied Space command)

BTW I still do not fully understand how the preview works, for some file types it displays the preview, but for others it displays nothing (or if the file does not have a file extension at all). Couldn't it display same content as the internal viewer in such cases?

**Comment by mikekazakov** [comment]
Nimble Commander does not do the previewing by itself, but instead it relies on the system-wide framework QuickLook to present the contents. To be able to show this additional information a separate logic for that needs to be implemented.
There's more info in the F.A.Q.: https://github.com/mikekazakov/nimble-commander/discussions/136
